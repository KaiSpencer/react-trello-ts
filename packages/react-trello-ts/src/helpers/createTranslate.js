import get from 'lodash/get'
export default (TABLE) => (key) => get(TABLE, key)
