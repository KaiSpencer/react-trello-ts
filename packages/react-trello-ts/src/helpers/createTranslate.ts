const getValue = (table) => (key) => {
	const keys = key.includes(".") ? key.split(".") : [key];
	let value = table;
	for (const k of keys) {
		if (value && k in value) {
			value = value[k];
		} else {
			return undefined;
		}
	}
	return value;
};

export default (table) => (key) => getValue(table)(key);
