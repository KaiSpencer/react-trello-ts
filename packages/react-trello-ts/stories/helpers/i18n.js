import i18n from "i18next";
import { initReactI18next } from "react-i18next";

import resources from 'rt/locales'

i18n
  .use(initReactI18next) // passes i18n down to react-i18next
  .init({
    resources,
    lng: "en"
  });

export default i18n;
