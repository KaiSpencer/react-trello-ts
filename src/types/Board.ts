import {CSSProperties} from 'react'

export interface BoardData {
  lanes: Lane[]
}

export interface Lane {
  id: string
  title?: string
  label?: string
  style?: CSSProperties
  cards: Card[]
  currentPage?: number
  droppable?: boolean
  labelStyle?: CSSProperties
  cardStyle?: CSSProperties
  disallowAddingCard?: boolean
  [key: string]: any
}

export interface Card {
  id: string
  title?: string
  label?: string
  description?: string
  laneId?: string
  style?: CSSProperties
  draggable?: boolean
  [key: string]: any
}
