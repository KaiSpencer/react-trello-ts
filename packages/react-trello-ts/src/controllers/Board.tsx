import React, { createContext, FC, PropsWithChildren } from "react";
import classNames from "classnames";
import uuidv1 from "uuid/v1";
import { BoardContainer } from "./BoardContainer";
import * as DefaultComponets from "../components";
import { BoardData } from "../types/Board";
import { store } from "../store/store";

export const Board: FC<
	PropsWithChildren<{
		id?: string;
		className?: string;
		components: typeof DefaultComponets;
		data: BoardData;
		t?: any;
	}>
> = ({ data, children, className, components, id, t, ...rest }) => {
	const allClassNames = classNames("react-trello-board", className || "");

	return (
		<BoardContext.Provider value={store}>
			<components.GlobalStyle />
			<BoardContainer
				id={id || uuidv1()}
				data={data}
				components={components}
				t={t}
				className={allClassNames}
				{...rest}
			/>
		</BoardContext.Provider>
	);
};

export const BoardContext = createContext(store);
