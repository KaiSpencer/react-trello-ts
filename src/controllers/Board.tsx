import React, {createContext, FC, PropsWithChildren} from 'react'
import classNames from 'classnames'
import uuidv1 from 'uuid/v1'
import {BoardContainer} from './BoardContainer'
import * as DefaultComponets from '../components'
import {BoardData} from 'rt/types/Board'
import {store} from 'rt/store/store'

// export class _Board extends Component<{
//   id?: string
//   className?: string
//   components?: typeof DefaultComponets
//   data: BoardData
//   t?: any
// }> {
//   store: any
//   id: any
//   constructor(props) {
//     super(props)
//     this.store = this.getStore()
//     this.id = this.props.id || uuidv1()
//   }

//   getStore = () => {
//     //When you create multiple boards, unique stores are created for isolation
//     return createStore(boardReducer)
//   }

//   render() {
//     const {className, components} = this.props
//     const allClassNames = classNames('react-trello-board', className || '')
//     return (
//       <Provider store={this.store}>
//         <>
//           <components.GlobalStyle />
//           <BoardContainer id={this.id} {...this.props} className={allClassNames} />
//         </>
//       </Provider>
//     )
//   }
// }

export const Board: FC<
  PropsWithChildren<{
    id?: string
    className?: string
    components: typeof DefaultComponets
    data: BoardData
    t?: any
  }>
> = ({data, children, className, components, id, t, ...rest}) => {
  const allClassNames = classNames('react-trello-board', className || '')

  return (
    <BoardContext.Provider value={store}>
      <components.GlobalStyle />
      <BoardContainer
        id={id || uuidv1()}
        data={data}
        components={components}
        t={t}
        className={allClassNames}
        {...rest}
      />
    </BoardContext.Provider>
  )
}

export const BoardContext = createContext(store)
