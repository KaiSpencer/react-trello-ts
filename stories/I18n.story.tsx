import { Meta, StoryObj } from "@storybook/react";
import React from "react";
import { I18nextProvider, useTranslation } from "react-i18next";

import Board from "../src";
import createTranslate from "../src/helpers/createTranslate";

import smallData from "./data/data-sort.json";
import i18n from "./helpers/i18n";
Board.displayName = "Board";

const I18nBoard = () => {
	const { t } = useTranslation();
	return (
		<div>
			<div>
				<button onClick={() => i18n.changeLanguage("en")}>English</button>
				<button onClick={() => i18n.changeLanguage("ru")}>Русский</button>
			</div>
			<Board
				data={smallData}
				t={t}
				editable={true}
				canAddLanes={true}
				draggable={true}
			/>
		</div>
	);
};

export default ({
	title: "I18n",
	component: Board,
} satisfies Meta<typeof Board>);
type Story = StoryObj<typeof Board>;
export const CustomTexts: Story = {
	name: "Custom texts",
	render: (args) => {
		const TEXTS = {
			"Add another lane": "NEW LANE",
			"Click to add card": "Click to add card",
			"Delete lane": "Delete lane",
			"Lane actions": "Lane actions",
			button: {
				"Add lane": "Add lane",
				"Add card": "Add card",
				Cancel: "Cancel",
			},
			placeholder: {
				title: "title",
				description: "description",
				label: "label",
			},
		};

		const customTranslation = createTranslate(TEXTS);
		return (
			<Board
				data={smallData}
				t={customTranslation}
				editable={true}
				canAddLanes={true}
				draggable={true}
			/>
		);
	},
};

export const FlatTranslationTable: Story = {
	name: "Flat translation table",
	render: (args) => {
		const FLAT_TRANSLATION_TABLE = {
			"Add another lane": "+ Weitere Liste erstellen",
			"Click to add card": "Klicken zum Erstellen einer Karte",
			"Delete lane": "Liste löschen",
			"Lane actions": "Listenaktionen",
			"button.Add lane": "Liste hinzufügen",
			"button.Add card": "Karte hinzufügen",
			"button.Cancel": "Abbrechen",
			"placeholder.title": "Titel",
			"placeholder.description": "Beschreibung",
			"placeholder.label": "Label",
		};

		return (
			<Board
				data={smallData}
				t={(key) => FLAT_TRANSLATION_TABLE[key]}
				editable={true}
				canAddLanes={true}
				draggable={true}
			/>
		);
	},
};

export const UsingI18next: Story = {
	name: "Using i18next",
	render: (args) => {
		return (
			<I18nextProvider i18n={i18n}>
				<I18nBoard />
			</I18nextProvider>
		);
	},
};
