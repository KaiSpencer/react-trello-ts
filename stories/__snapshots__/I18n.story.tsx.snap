// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`I18n CustomTexts smoke-test 1`] = `
<div draggable="false"
     class="sc-iGgWBj iSNfA"
>
  <div>
    <div class="smooth-dnd-container horizontal">
      <div class="smooth-dnd-draggable-wrapper">
        <section draggable="false"
                 class="sc-kAyceB dgWEjS react-trello-lane"
        >
          <header class="sc-gsFSXq sc-imWYAI imRIDz gLxuAH">
            <span class="sc-kpDqfm gDkBAH">
              Sorted Lane
            </span>
            <span class="sc-dAlyuH iHiWje">
              <span>
                20/70
              </span>
            </span>
          </header>
          <div class="sc-dhKdcB ikKiDc">
            <div class="smooth-dnd-container vertical">
              <div class="smooth-dnd-draggable-wrapper">
                <article data-id="Card1"
                         class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
                >
                  <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                    <span draggable="true"
                          class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                    >
                      <textarea placeholder="title"
                                autocomplete="off"
                                autocorrect="off"
                                autocapitalize="off"
                                spellcheck="false"
                                rows="1"
                                class="sc-fHjqPf bsHyvp"
                                style="overflow: hidden; overflow-wrap: break-word; text-align: start; height: 18px;"
                      >
                        Buy milk
                      </textarea>
                    </span>
                    <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                      <textarea placeholder="label"
                                autocomplete="off"
                                autocorrect="off"
                                autocapitalize="off"
                                spellcheck="false"
                                rows="1"
                                class="sc-fHjqPf bsHyvp"
                                style="overflow: hidden; overflow-wrap: break-word; text-align: right; height: 36px;"
                      >
                        2017-12-01
                      </textarea>
                    </span>
                    <div class="sc-hmdomO gJuJJJ">
                      <button class="sc-jsJBEP djrAfI">
                        ✖
                      </button>
                    </div>
                  </header>
                  <div class="sc-eDPEul cRhiim">
                    <textarea placeholder="description"
                              autocomplete="off"
                              autocorrect="off"
                              autocapitalize="off"
                              spellcheck="false"
                              rows="1"
                              class="sc-fHjqPf bsHyvp"
                              style="overflow: hidden; overflow-wrap: break-word; text-align: start; height: 36px;"
                    >
                      2 Gallons of milk at the Deli store
                    </textarea>
                  </div>
                </article>
              </div>
              <div class="smooth-dnd-draggable-wrapper">
                <article data-id="Card2"
                         class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
                >
                  <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                    <span draggable="true"
                          class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                    >
                      <textarea placeholder="title"
                                autocomplete="off"
                                autocorrect="off"
                                autocapitalize="off"
                                spellcheck="false"
                                rows="1"
                                class="sc-fHjqPf bsHyvp"
                                style="overflow: hidden; overflow-wrap: break-word; text-align: start; height: 18px;"
                      >
                        Dispose Garbage
                      </textarea>
                    </span>
                    <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                      <textarea placeholder="label"
                                autocomplete="off"
                                autocorrect="off"
                                autocapitalize="off"
                                spellcheck="false"
                                rows="1"
                                class="sc-fHjqPf bsHyvp"
                                style="overflow: hidden; overflow-wrap: break-word; text-align: right; height: 36px;"
                      >
                        2017-11-01
                      </textarea>
                    </span>
                    <div class="sc-hmdomO gJuJJJ">
                      <button class="sc-jsJBEP djrAfI">
                        ✖
                      </button>
                    </div>
                  </header>
                  <div class="sc-eDPEul cRhiim">
                    <textarea placeholder="description"
                              autocomplete="off"
                              autocorrect="off"
                              autocapitalize="off"
                              spellcheck="false"
                              rows="1"
                              class="sc-fHjqPf bsHyvp"
                              style="overflow: hidden; overflow-wrap: break-word; text-align: start; height: 36px;"
                    >
                      Sort out recyclable and waste as needed
                    </textarea>
                  </div>
                </article>
              </div>
              <div class="smooth-dnd-draggable-wrapper">
                <article data-id="Card3"
                         class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
                >
                  <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                    <span draggable="true"
                          class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                    >
                      <textarea placeholder="title"
                                autocomplete="off"
                                autocorrect="off"
                                autocapitalize="off"
                                spellcheck="false"
                                rows="1"
                                class="sc-fHjqPf bsHyvp"
                                style="overflow: hidden; overflow-wrap: break-word; text-align: start; height: 18px;"
                      >
                        Write Blog
                      </textarea>
                    </span>
                    <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                      <textarea placeholder="label"
                                autocomplete="off"
                                autocorrect="off"
                                autocapitalize="off"
                                spellcheck="false"
                                rows="1"
                                class="sc-fHjqPf bsHyvp"
                                style="overflow: hidden; overflow-wrap: break-word; text-align: right; height: 36px;"
                      >
                        2017-10-01
                      </textarea>
                    </span>
                    <div class="sc-hmdomO gJuJJJ">
                      <button class="sc-jsJBEP djrAfI">
                        ✖
                      </button>
                    </div>
                  </header>
                  <div class="sc-eDPEul cRhiim">
                    <textarea placeholder="description"
                              autocomplete="off"
                              autocorrect="off"
                              autocapitalize="off"
                              spellcheck="false"
                              rows="1"
                              class="sc-fHjqPf bsHyvp"
                              style="overflow: hidden; overflow-wrap: break-word; text-align: start; height: 18px;"
                    >
                      Can AI make memes?
                    </textarea>
                  </div>
                </article>
              </div>
              <div class="smooth-dnd-draggable-wrapper">
                <article data-id="Card4"
                         class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
                >
                  <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                    <span draggable="true"
                          class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                    >
                      <textarea placeholder="title"
                                autocomplete="off"
                                autocorrect="off"
                                autocapitalize="off"
                                spellcheck="false"
                                rows="1"
                                class="sc-fHjqPf bsHyvp"
                                style="overflow: hidden; overflow-wrap: break-word; text-align: start; height: 18px;"
                      >
                        Pay Rent
                      </textarea>
                    </span>
                    <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                      <textarea placeholder="label"
                                autocomplete="off"
                                autocorrect="off"
                                autocapitalize="off"
                                spellcheck="false"
                                rows="1"
                                class="sc-fHjqPf bsHyvp"
                                style="overflow: hidden; overflow-wrap: break-word; text-align: right; height: 36px;"
                      >
                        2017-09-01
                      </textarea>
                    </span>
                    <div class="sc-hmdomO gJuJJJ">
                      <button class="sc-jsJBEP djrAfI">
                        ✖
                      </button>
                    </div>
                  </header>
                  <div class="sc-eDPEul cRhiim">
                    <textarea placeholder="description"
                              autocomplete="off"
                              autocorrect="off"
                              autocapitalize="off"
                              spellcheck="false"
                              rows="1"
                              class="sc-fHjqPf bsHyvp"
                              style="overflow: hidden; overflow-wrap: break-word; text-align: start; height: 18px;"
                    >
                      Transfer to bank account
                    </textarea>
                  </div>
                </article>
              </div>
            </div>
            <a class="sc-gFqAkR ecfzPu">
              Click to add card
            </a>
          </div>
        </section>
      </div>
    </div>
  </div>
  <div class="smooth-dnd-container horizontal">
    <section class="sc-dAbbOL sc-feUZmu kUHSgc bMulzd">
      <button class="sc-fhzFiK duRxJO">
        NEW LANE
      </button>
    </section>
  </div>
</div>
`;

exports[`I18n FlatTranslationTable smoke-test 1`] = `
<div draggable="false"
     class="sc-iGgWBj iSNfA"
>
  <div>
    <div class="smooth-dnd-container horizontal">
      <div class="smooth-dnd-draggable-wrapper">
        <section draggable="false"
                 class="sc-kAyceB dgWEjS react-trello-lane"
        >
          <header class="sc-gsFSXq sc-imWYAI imRIDz gLxuAH">
            <span class="sc-kpDqfm gDkBAH">
              Sorted Lane
            </span>
            <span class="sc-dAlyuH iHiWje">
              <span>
                20/70
              </span>
            </span>
          </header>
          <div class="sc-dhKdcB ikKiDc">
            <div class="smooth-dnd-container vertical">
              <div class="smooth-dnd-draggable-wrapper">
                <article data-id="Card1"
                         class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
                >
                  <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                    <span draggable="true"
                          class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                    >
                      <textarea placeholder="Titel"
                                autocomplete="off"
                                autocorrect="off"
                                autocapitalize="off"
                                spellcheck="false"
                                rows="1"
                                class="sc-fHjqPf bsHyvp"
                                style="overflow: hidden; overflow-wrap: break-word; text-align: start; height: 18px;"
                      >
                        Buy milk
                      </textarea>
                    </span>
                    <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                      <textarea placeholder="Label"
                                autocomplete="off"
                                autocorrect="off"
                                autocapitalize="off"
                                spellcheck="false"
                                rows="1"
                                class="sc-fHjqPf bsHyvp"
                                style="overflow: hidden; overflow-wrap: break-word; text-align: right; height: 36px;"
                      >
                        2017-12-01
                      </textarea>
                    </span>
                    <div class="sc-hmdomO gJuJJJ">
                      <button class="sc-jsJBEP djrAfI">
                        ✖
                      </button>
                    </div>
                  </header>
                  <div class="sc-eDPEul cRhiim">
                    <textarea placeholder="Beschreibung"
                              autocomplete="off"
                              autocorrect="off"
                              autocapitalize="off"
                              spellcheck="false"
                              rows="1"
                              class="sc-fHjqPf bsHyvp"
                              style="overflow: hidden; overflow-wrap: break-word; text-align: start; height: 36px;"
                    >
                      2 Gallons of milk at the Deli store
                    </textarea>
                  </div>
                </article>
              </div>
              <div class="smooth-dnd-draggable-wrapper">
                <article data-id="Card2"
                         class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
                >
                  <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                    <span draggable="true"
                          class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                    >
                      <textarea placeholder="Titel"
                                autocomplete="off"
                                autocorrect="off"
                                autocapitalize="off"
                                spellcheck="false"
                                rows="1"
                                class="sc-fHjqPf bsHyvp"
                                style="overflow: hidden; overflow-wrap: break-word; text-align: start; height: 18px;"
                      >
                        Dispose Garbage
                      </textarea>
                    </span>
                    <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                      <textarea placeholder="Label"
                                autocomplete="off"
                                autocorrect="off"
                                autocapitalize="off"
                                spellcheck="false"
                                rows="1"
                                class="sc-fHjqPf bsHyvp"
                                style="overflow: hidden; overflow-wrap: break-word; text-align: right; height: 36px;"
                      >
                        2017-11-01
                      </textarea>
                    </span>
                    <div class="sc-hmdomO gJuJJJ">
                      <button class="sc-jsJBEP djrAfI">
                        ✖
                      </button>
                    </div>
                  </header>
                  <div class="sc-eDPEul cRhiim">
                    <textarea placeholder="Beschreibung"
                              autocomplete="off"
                              autocorrect="off"
                              autocapitalize="off"
                              spellcheck="false"
                              rows="1"
                              class="sc-fHjqPf bsHyvp"
                              style="overflow: hidden; overflow-wrap: break-word; text-align: start; height: 36px;"
                    >
                      Sort out recyclable and waste as needed
                    </textarea>
                  </div>
                </article>
              </div>
              <div class="smooth-dnd-draggable-wrapper">
                <article data-id="Card3"
                         class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
                >
                  <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                    <span draggable="true"
                          class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                    >
                      <textarea placeholder="Titel"
                                autocomplete="off"
                                autocorrect="off"
                                autocapitalize="off"
                                spellcheck="false"
                                rows="1"
                                class="sc-fHjqPf bsHyvp"
                                style="overflow: hidden; overflow-wrap: break-word; text-align: start; height: 18px;"
                      >
                        Write Blog
                      </textarea>
                    </span>
                    <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                      <textarea placeholder="Label"
                                autocomplete="off"
                                autocorrect="off"
                                autocapitalize="off"
                                spellcheck="false"
                                rows="1"
                                class="sc-fHjqPf bsHyvp"
                                style="overflow: hidden; overflow-wrap: break-word; text-align: right; height: 36px;"
                      >
                        2017-10-01
                      </textarea>
                    </span>
                    <div class="sc-hmdomO gJuJJJ">
                      <button class="sc-jsJBEP djrAfI">
                        ✖
                      </button>
                    </div>
                  </header>
                  <div class="sc-eDPEul cRhiim">
                    <textarea placeholder="Beschreibung"
                              autocomplete="off"
                              autocorrect="off"
                              autocapitalize="off"
                              spellcheck="false"
                              rows="1"
                              class="sc-fHjqPf bsHyvp"
                              style="overflow: hidden; overflow-wrap: break-word; text-align: start; height: 18px;"
                    >
                      Can AI make memes?
                    </textarea>
                  </div>
                </article>
              </div>
              <div class="smooth-dnd-draggable-wrapper">
                <article data-id="Card4"
                         class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
                >
                  <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                    <span draggable="true"
                          class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                    >
                      <textarea placeholder="Titel"
                                autocomplete="off"
                                autocorrect="off"
                                autocapitalize="off"
                                spellcheck="false"
                                rows="1"
                                class="sc-fHjqPf bsHyvp"
                                style="overflow: hidden; overflow-wrap: break-word; text-align: start; height: 18px;"
                      >
                        Pay Rent
                      </textarea>
                    </span>
                    <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                      <textarea placeholder="Label"
                                autocomplete="off"
                                autocorrect="off"
                                autocapitalize="off"
                                spellcheck="false"
                                rows="1"
                                class="sc-fHjqPf bsHyvp"
                                style="overflow: hidden; overflow-wrap: break-word; text-align: right; height: 36px;"
                      >
                        2017-09-01
                      </textarea>
                    </span>
                    <div class="sc-hmdomO gJuJJJ">
                      <button class="sc-jsJBEP djrAfI">
                        ✖
                      </button>
                    </div>
                  </header>
                  <div class="sc-eDPEul cRhiim">
                    <textarea placeholder="Beschreibung"
                              autocomplete="off"
                              autocorrect="off"
                              autocapitalize="off"
                              spellcheck="false"
                              rows="1"
                              class="sc-fHjqPf bsHyvp"
                              style="overflow: hidden; overflow-wrap: break-word; text-align: start; height: 18px;"
                    >
                      Transfer to bank account
                    </textarea>
                  </div>
                </article>
              </div>
            </div>
            <a class="sc-gFqAkR ecfzPu">
              Klicken zum Erstellen einer Karte
            </a>
          </div>
        </section>
      </div>
    </div>
  </div>
  <div class="smooth-dnd-container horizontal">
    <section class="sc-dAbbOL sc-feUZmu kUHSgc bMulzd">
      <button class="sc-fhzFiK duRxJO">
        + Weitere Liste erstellen
      </button>
    </section>
  </div>
</div>
`;

exports[`I18n UsingI18next smoke-test 1`] = `
<div>
  <div>
    <button>
      English
    </button>
    <button>
      Русский
    </button>
  </div>
  <div draggable="false"
       class="sc-iGgWBj iSNfA"
  >
    <div>
      <div class="smooth-dnd-container horizontal">
        <div class="smooth-dnd-draggable-wrapper">
          <section draggable="false"
                   class="sc-kAyceB dgWEjS react-trello-lane"
          >
            <header class="sc-gsFSXq sc-imWYAI imRIDz gLxuAH">
              <span class="sc-kpDqfm gDkBAH">
                Sorted Lane
              </span>
              <span class="sc-dAlyuH iHiWje">
                <span>
                  20/70
                </span>
              </span>
            </header>
            <div class="sc-dhKdcB ikKiDc">
              <div class="smooth-dnd-container vertical">
                <div class="smooth-dnd-draggable-wrapper">
                  <article data-id="Card1"
                           class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
                  >
                    <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                      <span draggable="true"
                            class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                      >
                        <textarea placeholder="title"
                                  autocomplete="off"
                                  autocorrect="off"
                                  autocapitalize="off"
                                  spellcheck="false"
                                  rows="1"
                                  class="sc-fHjqPf bsHyvp"
                                  style="overflow: hidden; overflow-wrap: break-word; text-align: start; height: 18px;"
                        >
                          Buy milk
                        </textarea>
                      </span>
                      <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                        <textarea placeholder="label"
                                  autocomplete="off"
                                  autocorrect="off"
                                  autocapitalize="off"
                                  spellcheck="false"
                                  rows="1"
                                  class="sc-fHjqPf bsHyvp"
                                  style="overflow: hidden; overflow-wrap: break-word; text-align: right; height: 36px;"
                        >
                          2017-12-01
                        </textarea>
                      </span>
                      <div class="sc-hmdomO gJuJJJ">
                        <button class="sc-jsJBEP djrAfI">
                          ✖
                        </button>
                      </div>
                    </header>
                    <div class="sc-eDPEul cRhiim">
                      <textarea placeholder="description"
                                autocomplete="off"
                                autocorrect="off"
                                autocapitalize="off"
                                spellcheck="false"
                                rows="1"
                                class="sc-fHjqPf bsHyvp"
                                style="overflow: hidden; overflow-wrap: break-word; text-align: start; height: 36px;"
                      >
                        2 Gallons of milk at the Deli store
                      </textarea>
                    </div>
                  </article>
                </div>
                <div class="smooth-dnd-draggable-wrapper">
                  <article data-id="Card2"
                           class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
                  >
                    <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                      <span draggable="true"
                            class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                      >
                        <textarea placeholder="title"
                                  autocomplete="off"
                                  autocorrect="off"
                                  autocapitalize="off"
                                  spellcheck="false"
                                  rows="1"
                                  class="sc-fHjqPf bsHyvp"
                                  style="overflow: hidden; overflow-wrap: break-word; text-align: start; height: 18px;"
                        >
                          Dispose Garbage
                        </textarea>
                      </span>
                      <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                        <textarea placeholder="label"
                                  autocomplete="off"
                                  autocorrect="off"
                                  autocapitalize="off"
                                  spellcheck="false"
                                  rows="1"
                                  class="sc-fHjqPf bsHyvp"
                                  style="overflow: hidden; overflow-wrap: break-word; text-align: right; height: 36px;"
                        >
                          2017-11-01
                        </textarea>
                      </span>
                      <div class="sc-hmdomO gJuJJJ">
                        <button class="sc-jsJBEP djrAfI">
                          ✖
                        </button>
                      </div>
                    </header>
                    <div class="sc-eDPEul cRhiim">
                      <textarea placeholder="description"
                                autocomplete="off"
                                autocorrect="off"
                                autocapitalize="off"
                                spellcheck="false"
                                rows="1"
                                class="sc-fHjqPf bsHyvp"
                                style="overflow: hidden; overflow-wrap: break-word; text-align: start; height: 36px;"
                      >
                        Sort out recyclable and waste as needed
                      </textarea>
                    </div>
                  </article>
                </div>
                <div class="smooth-dnd-draggable-wrapper">
                  <article data-id="Card3"
                           class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
                  >
                    <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                      <span draggable="true"
                            class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                      >
                        <textarea placeholder="title"
                                  autocomplete="off"
                                  autocorrect="off"
                                  autocapitalize="off"
                                  spellcheck="false"
                                  rows="1"
                                  class="sc-fHjqPf bsHyvp"
                                  style="overflow: hidden; overflow-wrap: break-word; text-align: start; height: 18px;"
                        >
                          Write Blog
                        </textarea>
                      </span>
                      <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                        <textarea placeholder="label"
                                  autocomplete="off"
                                  autocorrect="off"
                                  autocapitalize="off"
                                  spellcheck="false"
                                  rows="1"
                                  class="sc-fHjqPf bsHyvp"
                                  style="overflow: hidden; overflow-wrap: break-word; text-align: right; height: 36px;"
                        >
                          2017-10-01
                        </textarea>
                      </span>
                      <div class="sc-hmdomO gJuJJJ">
                        <button class="sc-jsJBEP djrAfI">
                          ✖
                        </button>
                      </div>
                    </header>
                    <div class="sc-eDPEul cRhiim">
                      <textarea placeholder="description"
                                autocomplete="off"
                                autocorrect="off"
                                autocapitalize="off"
                                spellcheck="false"
                                rows="1"
                                class="sc-fHjqPf bsHyvp"
                                style="overflow: hidden; overflow-wrap: break-word; text-align: start; height: 18px;"
                      >
                        Can AI make memes?
                      </textarea>
                    </div>
                  </article>
                </div>
                <div class="smooth-dnd-draggable-wrapper">
                  <article data-id="Card4"
                           class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
                  >
                    <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                      <span draggable="true"
                            class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                      >
                        <textarea placeholder="title"
                                  autocomplete="off"
                                  autocorrect="off"
                                  autocapitalize="off"
                                  spellcheck="false"
                                  rows="1"
                                  class="sc-fHjqPf bsHyvp"
                                  style="overflow: hidden; overflow-wrap: break-word; text-align: start; height: 18px;"
                        >
                          Pay Rent
                        </textarea>
                      </span>
                      <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                        <textarea placeholder="label"
                                  autocomplete="off"
                                  autocorrect="off"
                                  autocapitalize="off"
                                  spellcheck="false"
                                  rows="1"
                                  class="sc-fHjqPf bsHyvp"
                                  style="overflow: hidden; overflow-wrap: break-word; text-align: right; height: 36px;"
                        >
                          2017-09-01
                        </textarea>
                      </span>
                      <div class="sc-hmdomO gJuJJJ">
                        <button class="sc-jsJBEP djrAfI">
                          ✖
                        </button>
                      </div>
                    </header>
                    <div class="sc-eDPEul cRhiim">
                      <textarea placeholder="description"
                                autocomplete="off"
                                autocorrect="off"
                                autocapitalize="off"
                                spellcheck="false"
                                rows="1"
                                class="sc-fHjqPf bsHyvp"
                                style="overflow: hidden; overflow-wrap: break-word; text-align: start; height: 18px;"
                      >
                        Transfer to bank account
                      </textarea>
                    </div>
                  </article>
                </div>
              </div>
              <a class="sc-gFqAkR ecfzPu">
                Click to add card
              </a>
            </div>
          </section>
        </div>
      </div>
    </div>
    <div class="smooth-dnd-container horizontal">
      <section class="sc-dAbbOL sc-feUZmu kUHSgc bMulzd">
        <button class="sc-fhzFiK duRxJO">
          + Add another lane
        </button>
      </section>
    </div>
  </div>
</div>
`;
